File: algorithms/1_easy/F/02239_find_closest_number_to_zero/solution.cpp
class Solution {
public:
    int findClosestNumber(vector<int>& nums) {
        int res{std::numeric_limits<int>::max()};

        for (const int &num: nums) {
            if (abs(num) < abs(res)) {
                res = num;
            } else if (abs(num) == abs(res)) {
                res = max(res, num);
            }
        }

        return res;
    }
};

File: algorithms/1_easy/F/02239_find_closest_number_to_zero/solution.rs
use std::cmp::Ordering;

impl Solution {
    pub fn find_closest_number(nums: Vec<i32>) -> i32 {
        let mut res = i32::MAX;

        for num in nums.into_iter() {
            match num.abs().cmp(&res.abs()) {
                Ordering::Greater => {},
                Ordering::Equal => res = res.max(num),
                Ordering::Less => res = num,
            }
        }

        res
    }
}
